src/emailservice/email_server.rb

```diff
@@ -0,0 +1,22 @@
+require "ostruct"
+require "pony"
+require "sinatra"
+
+require "opentelemetry/sdk"
+require "opentelemetry/exporter/otlp"
+require "opentelemetry/instrumentation/sinatra"
+
+OpenTelemetry::SDK.configure do |c|
+  c.use "OpenTelemetry::Instrumentation::Sinatra"
+end
+
+post "/send_order_confirmation" do
+  data = JSON.parse(request.body.read, object_class: OpenStruct)
```

Comment:
Any change we can use a Struct or Object here instead to mitigate memory leaks?

Reply:
Feel free to open a pr with the proposed changes. I know @ahayworth would love the help!